### DISCUSSION TITLE
Rolling update with Docker Compose

### DISCUSSION THREAD

**Discussion by softwaretirol** [coollabsio/coolify#2824] [Q&A]
Is it possible to perform a rolling update with a Docker Compose file? It works well with a simple Docker image, but it doesn't seem to work with a Docker Compose file.

**Comment by terijaki** [comment]
**_As far as I understand it,_** rolling updates work as long as you do not set custom container names.
At least that's what the UI communicates under `Configuration > Advanced`

![image](https://github.com/user-attachments/assets/fdf5d49c-6d56-484d-b494-2a8f4c66b0cc)

**Comment by terijaki** [comment]
I found that coolify is adding a container name to the compose content:
<img width="765" alt="image" src="https://github.com/user-attachments/assets/a861491a-7449-488a-86c3-7c7f8bd1f4b4">

Probably related to making the proxy work.

This is preventing rolling updates and replicas.

**Reply by Thytu** [reply]
Yes, most probably linked to #2131

**Comment by vanWittlaer** [comment]
As I am facing same problem (no rolling update with my docker compose setup) I wonder shouldn't we make this an issue?

**Reply by Thytu** [reply]
Sure, go ahead! You could link both this conversation and the issue #2131.

**Reply by vanWittlaer** [reply]
I have created an issue https://github.com/coollabsio/coolify/issues/3408 - it might not be related, i.e. different root causes (container name vs. image name).
